simperby/src/node.rs

```diff
@@ -0,0 +1,196 @@
+mod storage;
+
+use super::*;
+use simperby_kv_storage::KVStorage;
+use simperby_network::AuthorizedNetwork;
+use std::sync::Arc;
+use storage::*;
+use tokio::sync::RwLock;
+use vetomint::Round;
+
+pub struct Node {
+    state: Arc<RwLock<NodeState>>,
+    network_task: tokio::task::JoinHandle<()>,
+    genesis_info: GenesisInfo,
+}
+
+impl Node {
+    pub(crate) async fn new(
+        genesis_info: GenesisInfo,
+        network: Box<dyn AuthorizedNetwork>,
+        state_storage: Box<dyn KVStorage>,
+        history_storage: Box<dyn KVStorage>,
+    ) -> Result<Self, anyhow::Error> {
+        let last_header = genesis_info.header.clone();
+        let state_ = Arc::new(RwLock::new(NodeState {
```

Comment:
Is this trailing underscore for the implication of wrapping `Arc`?

Reply:
Yes but not always. A trailing underscore generally means a temporary variable that is related to the original name.